main.go

```diff
@@ -46,8 +45,42 @@ import (
 	"github.com/prometheus/pushgateway/storage"
 )
 
+var (
+	requestDuration = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "pushgateway_http_request_duration_seconds",
+			Help:    "Histogram of latencies for HTTP requests.",
+			Buckets: []float64{.05, 0.1, .25, .5, .75, 1, 2, 5, 20, 60},
+		},
+		[]string{"handler", "method"},
+	)
+	responseSize = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "pushgateway_http_response_size_bytes",
+			Help:    "Histogram of response size for HTTP requests.",
+			Buckets: prometheus.ExponentialBuckets(100, 10, 7),
+		},
+		[]string{"handler", "method"},
```

Comment:
These are good metrics, but I think they are overkill for the Pushgateway. Or let's say I would only introduce this level of detail if somebody asks for it.

Also, these metrics compete now with the metrics in handler/metrics.go.

Let's for now only use the metrics we already have. I'd suggest to leave `httpPushDuration` and `httpPushDuration` exactly as they are, with the explicit handling in the push handler, and only use `httpCnt` in a generic way.